### PR TITLE
Move vuln scan action from agent repo.

### PR DESCRIPTION
**Pull request by purple4reina**
In order to be more proactive on vulnerabilities in the serverless extension, run a vulnerability scan nightly on the agent `main` branch.

We will be able to subscribe in slack to this workflow which I will do once we're done testing it.

### CODE REVIEW

File: .github/workflows/serverless-vuln-scan.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+env:
+  VERSION: 1  # env var required when building extension
+
+jobs:
+  run:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout datadog-agent repository
+        uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4.1.4
```
Comment (duncanista) [inline .github/workflows/serverless-vuln-scan.yml:16]: Why are we adding hashes instead of the version?

Comment (purple4reina) [reply .github/workflows/serverless-vuln-scan.yml:16]: I copied this from the agent repo. It was changed to hashes for security reasons.

**Comment by purple4reina** [comment]
Also note, this doesn't scan serverless-init or bottlecap. We can always add those later.

**Review by DylanLovesCoffee — APPROVED** [review]
🥹

**Pull request merged**